huaweicloud/services/acceptance/dcs/resource_huaweicloud_dcs_backup_test.go

```diff
@@ -60,20 +51,19 @@ func getDcsBackupResourceFunc(cfg *config.Config, state *terraform.ResourceState
 		total := utils.PathSearch("total_num", getBackupRespBody, 0)
 		for _, backup := range backups {
 			id := utils.PathSearch("backup_id", backup, "")
-			if id != backupId {
+			if id != state.Primary.ID {
 				continue
 			}
 			status := utils.PathSearch("status", backup, "")
 			if status == "deleted" {
-				return nil, fmt.Errorf("error get DCS backup by backup_id (%s)", backupId)
+				return nil, fmt.Errorf("error get DCS backup by backup_id (%s)", state.Primary.ID)
 			}
 			return backup, nil
 		}
 		currentTotal += len(backups)
-		if currentTotal == total {
+		if float64(currentTotal) == total {
```

Comment:
Will there be any problems if we directly compare decimals?

Reply:
done